README.md

```diff
@@ -7,290 +7,505 @@
-- Now the SDK supports RS256 codes, it will decode using the `.well-known/jwks.json` endpoint to fetch the public key
+See the "Usage - Management API" section below for more information on how to use this access token. 
 
-#### 2.x
+## Usage - Decoding and Verifying JWTs
 
-- Session storage now returns null (and null is expected by the sdk) if there is no info stored (this change was made since false is a valid value to be stored in session).
-- Guzzle 6.1 required
+This SDK also includes an interface to the [Firebase PHP JWT library](https://github.com/firebase/php-jwt), used to decode and verify JSON web tokens (JWT). The `JWTVerifier` class has a single method, `verifyAndDecode()`, which accepts a JWT and either returns a decoded token or throws an error. More information on JWTs and how to build and decode them can be found [here on jwt.io](https://jwt.io/).
 
-#### 1.x
+The decoder can work with both HS256 and RS256 tokens. Both types require the algorithm and valid audiences to indicated before processing. Additionally, HS256 tokens require the client secret while RS256 tokens require an authorized issuer. 
 
-- Now, all the SDK is under the namespace `\Auth0\SDK`
-- The exceptions were moved to the namespace `\Auth0\SDK\Exceptions`
+Here is an example of a small, URL-based JWT decoder:
 
-### New features
 
-- The Auth0 class, now provides two methods to access the user metadata, `getUserMetadata` and `getAppMetadata`. For more info, check the [API v2 changes](https://auth0.com/docs/apiv2Changes)
-- The Auth0 class, now provides a way to update the UserMetadata with the method `updateUserMetadata`. Internally, it uses the [update user endpoint](https://auth0.com/docs/apiv2#!/users/patch_users_by_id), check the method documentation for more info.
-- The new service `\Auth0\SDK\API\ApiUsers` provides an easy way to consume the API v2 Users endpoints.
-- A simple API client (`\Auth0\SDK\API\ApiClient`) is also available to use.
-- A JWT generator and decoder is also available (`\Auth0\SDK\Auth0JWT`)
-- Now provides an interface for the [Authentication API](https://auth0.com/docs/auth-api).
+```php
+// Example #4
+// decode-jwt.php
+use Auth0\SDK\JWTVerifier;
+use Auth0\SDK\Exception\InvalidTokenException;
+use Auth0\SDK\Exception\CoreException;
 
->***Note:*** API V2 restrict the access to the endpoints via scopes. By default, the user token has granted certain scopes that let update the user metadata but not the root attributes nor app_metadata. To update this information and access another endpoints, you need an special token with this scopes granted. For more information about scopes, check [the API v2 docs](https://auth0.com/docs/apiv2Changes#6).
+// Do we have an ID token?
+if (empty($_GET[ 'id_token' ])) {
+    echo '<code>No `id_token` URL parameter!</code> ';
+    exit;
+}
+
+// Do we have a valid algorithm?
+if (empty($_GET[ 'token_alg' ]) || ! in_array($_GET[ 'token_alg' ], [ 'HS256', 'RS256' ])) {
```

Comment:
Can we discuss how this works? It's weird to take all those params from the URL (query I guess?).

Reply:
It's just an example, easier way to show data coming from somewhere in a way that's easy to test my example. If this is overly complex, I can just use, like, `JWT_SIGNING_ALGO` and `get_id_token()` or something